federation/apis/federation/v1alpha1/types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// ClusterSpec describes the attributes on a Cluster.
+type ClusterSpec struct {
+	// a map of client CIDR to server address
+	ServerAddressByClientCIDRs []unversioned.ServerAddressByClientCIDR `json:"serverAddressByClientCIDRs"`
+	// the type (e.g. bearer token, client certificate etc) and data of the credential used to access cluster.
+	// It’s used for system routines (not behalf of users)
+	Credential string `json:"credential",omitempty`
```

Comment:
It's in https://github.com/quinton-hoole/kubernetes/blob/2015-01-05-ube-design-docs/docs/design/federation-phase-1.md
@quinton-hoole